addons/controllers/packageinstallstatus_controller.go

```diff
@@ -0,0 +1,448 @@
+}
+
+// watchPackageInstalls sets a remote watch on the provided cluster on the Kind resource
+func (r *PackageInstallStatusReconciler) watchPackageInstalls(ctx context.Context, cluster *clusterapiv1beta1.Cluster, log logr.Logger) error {
+	// If there is no tracker, don't watch remote package installs
+	if r.Tracker == nil {
+		return nil
+	}
+
+	return r.Tracker.Watch(ctx, remote.WatchInput{
+		Name:         "watchPackageInstallStatus",
+		Cluster:      clusterapiutil.ObjectKey(cluster),
+		Watcher:      r.controller,
+		Kind:         &kapppkgiv1alpha1.PackageInstall{},
+		EventHandler: handler.EnqueueRequestsFromMapFunc(r.pkgiToCluster),
+		// ClusterBootstrap resource only exists in the management cluster, hence using local client
+		Predicates: []predicate.Predicate{r.PkgiStatusChanged(log), predicates.ClusterHasLabel(constants.TKRLabelClassyClusters, r.Log)},
+	})
+}
+
+// SetupWithManager sets up the controller with the Manager.
+func (r *PackageInstallStatusReconciler) SetupWithManager(ctx context.Context, mgr ctrl.Manager, options controller.Options) error {
+	pkgiStatusController, err := ctrl.NewControllerManagedBy(mgr).
+		For(&clusterapiv1beta1.Cluster{}).
+		Watches(
+			&source.Kind{Type: &kapppkgiv1alpha1.PackageInstall{}},
+			handler.EnqueueRequestsFromMapFunc(r.pkgiToCluster),
+			builder.WithPredicates(r.PkgiStatusChanged(r.Log)),
+		).
+		WithOptions(options).
+		WithEventFilter(predicates.ClusterHasLabel(constants.TKRLabelClassyClusters, r.Log)).
+		Build(r)
+
+	if err != nil {
+		return errors.Wrap(err, "failed setting up with a controller manager")
+	}
+
+	r.controller = pkgiStatusController
+	r.Ctx = ctx
+	if r.aggregatedAPIResourcesClient, err = client.New(mgr.GetConfig(), client.Options{Scheme: mgr.GetScheme()}); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// pkgiToCluster returns a list of Requests with Cluster ObjectKey
+func (r *PackageInstallStatusReconciler) pkgiToCluster(o client.Object) []ctrl.Request {
+	pkgi, ok := o.(*kapppkgiv1alpha1.PackageInstall)
+	if !ok {
+		r.Log.Error(errors.New("invalid type"),
+			"Expected to receive PackageInstall resource",
+			"actualType", fmt.Sprintf("%T", o))
+		return nil
+	}
+
+	r.Log.WithValues("pkgi-name", pkgi.Name).Info("Mapping PackageInstalls to cluster")
+
+	clusterNamespace, clusterName := r.getClusterNamespacedName(pkgi)
+	if clusterNamespace == "" || clusterName == "" {
+		return nil
+	}
+
+	// TODO: check if the following check is needed here considering that we check it in PkgiStatusChanged predicate
+	isManaged, err := r.isPackageManaged(clusterName, clusterNamespace, pkgi.Name)
+	if err != nil || !isManaged {
+		return nil
+	}
+
+	return []ctrl.Request{{NamespacedName: client.ObjectKey{Namespace: clusterNamespace, Name: clusterName}}}
+}
+
+// PkgiStatusChanged returns a predicate.Predicate that filters pkgi objects which their status has changed
+func (r *PackageInstallStatusReconciler) PkgiStatusChanged(log logr.Logger) predicate.Funcs {
```

Comment:
Maybe consider renaming the func/predicate as it checks more than just status changed.

Reply:
@sbueringer It checks 1) it be a PackageInstall resource and 2) It be managed. Does ```managedPkgiStatusChanged``` sound better to you?:)

Reply:
Following CAPI patterns we would probably call it `PkgiIsManagedAndStatusChanged`

A bit more explicit but I consider this a feature to be clearer here :)

Just noticed, the `predicates.ClusterHasLabel(constants.TKRLabelClassyClusters, log)` predicate won't work here as the predicates are applied to the objects we're watching (i.e. PackageInstall) so this predicate will just be a no-op.

Reply:
Thanks for the suggestion, changed. 
Oh yes. You were indeed right that ```ClusterHasLabel``` would be no-op inside  ```watchPackageInstalls```; Removed.